### ISSUE TITLE
example

### ISSUE THREAD

**Issue by ai8049520** [prymeka/mosaic-augmentation-pytorch#1]
Hello!  Can you give example for your Mosaic implementation?
#dataset

class PennFudanDataset(torch.utils.data.Dataset):
    def __init__(self, split, root="experiment/dataset/fgp", train_transform=get_transform):
        self.root = root
        self.transforms = train_transform(split)

        # Load all image and mask filenames
        self.imgs = list(sorted(os.listdir(os.path.join(root, "PNGimages"))))
        self.masks = list(sorted(os.listdir(os.path.join(root, "masks"))))

        # Shuffle indices once to ensure randomness
        random.seed(SEED)
        indices = list(range(len(self.imgs)))
        random.shuffle(indices)

        # Apply slicing based on the split
        if split == "train":
            start, end = 0, int(0.8 * len(self.imgs))
        elif split in ["val", "test", "inference"]:
            start, end = int(0.8 * len(self.imgs)), len(self.imgs)
        else:
            raise ValueError(f"Invalid split: {split}")

        # Slice the dataset
        self.imgs = [self.imgs[i] for i in indices[start:end]]
        self.masks = [self.masks[i] for i in indices[start:end]]

    def __getitem__(self, idx):
        # Load image and mask as before
        img_path = os.path.join(self.root, "PNGimages", self.imgs[idx])
        mask_path = os.path.join(self.root, "masks", self.masks[idx])
        img = read_image(img_path)
        mask = read_image(mask_path).to(dtype=torch.uint8)

        # Process targets as before
        obj_ids = torch.unique(mask)
        obj_ids = obj_ids[1:]  # Exclude background
        masks = (mask == obj_ids[:, None, None]).to(dtype=torch.uint8)
        boxes = masks_to_boxes(masks)
        labels = torch.ones((len(obj_ids),), dtype=torch.int64)

        target = {
            "boxes": tv_tensors.BoundingBoxes(boxes, format="XYXY", canvas_size=FT.get_image_size(img)),
            "masks": tv_tensors.Mask(masks),
            "labels": labels,
        }

        if self.transforms is not None:
            img, target = self.transforms(img, target)

        return img, target

    def __len__(self):
        return len(self.imgs)

from torchvision.transforms import v2 as VT

def get_transform(split):
    transforms = []
    if split == "train":
        transforms.append(VT.RandomHorizontalFlip(0.5))
        transforms.append(VT.RandomIoUCrop(min_scale= 0.3, max_scale = 1.0, min_aspect_ratio = 0.5, max_aspect_ratio = 2.0))
        transforms.append(VT.RandomZoomOut(fill = 0 ))
        transforms.append(VT.RandomPhotometricDistort(p=0.5))
        transforms.append(VT.ColorJitter(brightness=0.5, contrast=0.5, saturation=0.5, hue=0.2))
        transforms.append(VT.SanitizeBoundingBoxes(min_size=1))
    transforms.append(VT.ToDtype(torch.float, scale=True))
    transforms.append(VT.ToPureTensor())
    return VT.Compose(transforms)

This is currently my working schema. How can I implement Mosaic class, wich extends it?

**Comment by prymeka** [comment]
Hi, mosaic can not be used like typical Torchivision transforms as the input is expected to be a batch of samples. As explained in README, mosaic should be applied inside the training loop and not as a part of a data loader - just like Torchivision's `CutMix` or `MixUp` transformations. 

You can check out how to use `CutMix`/`MixUp` transformations (which use is identical to the mosaic transform) [here](https://pytorch.org/vision/main/auto_examples/transforms/plot_cutmix_mixup.html) or at the bottom of README.
